test/framework/clusterctl/e2e_config.go

```diff
@@ -66,15 +70,11 @@ func LoadE2EConfig(ctx context.Context, input LoadE2EConfigInput) *E2EConfig {
 
 // SetCNIEnvVar read CNI from cniManifestPath and sets an environmental variable that keeps CNI resources.
 // A ClusterResourceSet can be used to apply CNI using this environmental variable.
+//
+// Deprecated: CNI manifest will be now automatically embedded into cluster templates during create repository.
+// The new approach does not uses env variables so we can avoid https://github.com/kubernetes-sigs/cluster-api/issues/3797;
+// This func is preserved for avoiding to break users in the v0.3 series, but it is now a noop.
 func SetCNIEnvVar(cniManifestPath string, cniEnvVar string) {
```

Comment:
should we make this PR has "breaking" since this is now a no-op?

Reply:
makes sense, done + updated PR comment